torch/_inductor/triton_heuristics.py

```diff
@@ -67,6 +84,8 @@ def __init__(self, fn, meta, configs, save_cache_hook, mutated_arg_names):
                 str(self.meta.get("device", 0)),
             )
 
+        self.coordesc_tuner = CoordescTuner(is_mm=False, is_persistent_reduction=self.is_persistent_reduction())
```

Comment:
btw, I think we should just modify codegen for persistent reductions so that RBLOCK isn't a kernel parameter.

Reply:
We also need remove RBLOCK from persistent reduction's triton.Config? Since the coordinate descent tuning looks for all the fields in a triton.Config and try to tune them.

Reply:
Yeah we should just make the changes to use rnumel instead of RBLOCK and not make it a tunable parameter in the config.

Reply:
I just found RBLOCK is a power of 2 while rnumel may be not. E.g.
```
@triton.jit
def triton_(in_out_ptr0, in_ptr0, in_ptr1, in_ptr2, in_ptr3, in_ptr4, in_ptr5, in_ptr6, in_ptr7, out_ptr2, xnumel, rnumel, XBLOCK : tl.constexpr, RBLOCK : tl.constexpr):
    xnumel = 4096
    rnumel = 768
```

Should we be worried about that? Not sure if this impact perf.

Reply:
To be safe, should we generated code like:
```
@persistent_reduction(
    size_hints=[4096, 1024],
    reduction_hint=ReductionHint.INNER,
    filename=__file__,
    meta={'signature': {0: '*fp32', 1: '*i64', 2: '*fp16', 3: '*i64', 4: '*fp16', 5: '*i64', 6: '*fp16', 7: '*fp16', 8: '*fp16', 9: '*fp16', 10: 'i32', 11: 'i32'}, 'device': 0, 'constants': {}, 'mutated_arg_names': ['in_out_ptr0'], 'configs': [instance_descriptor(divisible_by_16=(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11), equal_to_1=())]}
)
@triton.jit
def triton_(in_out_ptr0, in_ptr0, in_ptr1, in_ptr2, in_ptr3, in_ptr4, in_ptr5, in_ptr6, in_ptr7, out_ptr2, xnumel, rnumel, XBLOCK : tl.constexpr):
    xnumel = 4096
    rnumel = 768
    RBLOCK = 1024 # round rnumel to the next power of 2
```

cc @Chillee

Reply:
Yes for persistent reduction we should codegen RBLOCK that's the next power of 2.

Reply:
I resolve this comment in a separate PR here https://github.com/pytorch/pytorch/pull/98653 .